source/common/stats/stat_merger.cc

```diff
@@ -32,23 +49,16 @@ void StatMerger::mergeGauges(const Protobuf::Map<std::string, uint64_t>& gauges)
       }
     }
 
-    // We establish here a tentative import-mode of Uninitialized. Gauges in
-    // this mode will not be reported in stats sinks or in the admin /stats
-    // endpoint. However, we'll retain the transferred value, and if the running
-    // system initialized the stat as Accumulate, we'll have the accumulated
-    // value ready to go. If the system re-initializes it as NeverImport, we'll
-    // clear the value during the mergeImportMode call.
     auto& gauge_ref = temp_scope_->gaugeFromStatName(stat_name, import_mode);
     uint64_t& parent_value_ref = parent_gauge_values_[gauge_ref.statName()];
     uint64_t old_parent_value = parent_value_ref;
     uint64_t new_parent_value = gauge.second;
     parent_value_ref = new_parent_value;
 
-    if (new_parent_value > old_parent_value) {
-      gauge_ref.add(new_parent_value - old_parent_value);
-    } else {
-      gauge_ref.sub(old_parent_value - new_parent_value);
-    }
+    // Note that new_parent_value may be less than old_parent_value, in which
```

Comment:
Do we have sufficient tests for this wrapping?

Reply:
I think so; one is:
```
TEST_F(StatMergerTest, multipleImportsWithAccumulationLogic) {
  {
    Protobuf::Map<std::string, uint64_t> gauges;
    gauges["whywassixafraidofseven"] = 100;
    stat_merger_.mergeStats(empty_counter_deltas_, gauges);
    // Initial combined values: 678+100 and 1+2.
    EXPECT_EQ(778, whywassixafraidofseven_.value());
  }
  {
    Protobuf::Map<std::string, uint64_t> gauges;
    // The parent's gauge drops by 1, and its counter increases by 1.
    gauges["whywassixafraidofseven"] = 99;
    stat_merger_.mergeStats(empty_counter_deltas_, gauges);
    EXPECT_EQ(777, whywassixafraidofseven_.value());
```

dropping whywassixafraidofseven from 100 to 99 triggers that behavior.